woocommerce/woocommerce
# Fix flaky orphaned refund test

Make sure the date_paid -> date -> order is deterministic.

Fixes https://github.com/woocommerce/woocommerce/issues/49630

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR fixes the flaky orphaned refunds test. When the order and the refund are created, it's possible they both have the same `paid_date` -> `date`, which makes the order of them in the report results random. Therefore, asserting the first or second may produce different outcomes.

Closes https://github.com/woocommerce/woocommerce/issues/49630
Reverts https://github.com/woocommerce/woocommerce/pull/49711



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This is a developer-facing change.

1. Checkout the branch
2. Start test environment `pnpm --filter="@woocommerce/plugin-woocommerce" test:php:env`
3. Run PHP unit tests `pnpm --filter="@woocommerce/plugin-woocommerce" env:test`
4. Re-run the suite or the single test a few times more to make sure it consistently passes `pnpm --filter="@woocommerce/plugin-woocommerce" test:php:env -- --filter=test_get_reports_with_orphaned_refund`



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Fix flaky orphaned refund test

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
